vfs/s3fs.go

```diff
@@ -138,51 +137,14 @@ func (fs *S3Fs) Stat(name string) (os.FileInfo, error) {
 	}
 	// the requested file could still be a directory as a zero bytes key
 	// with a forwarding slash.
-	// As last resort we do a list dir to find it
-	return fs.getStatCompat(name)
-}
-
-func (fs *S3Fs) getStatCompat(name string) (os.FileInfo, error) {
-	var result FileInfo
-	prefix := path.Dir(name)
-	if prefix == "/" || prefix == "." {
-		prefix = ""
-	} else {
-		prefix = strings.TrimPrefix(prefix, "/")
-		if !strings.HasSuffix(prefix, "/") {
-			prefix += "/"
-		}
+	obj, err = fs.headObject(strings.TrimPrefix(name, "/") + "/")
```

Comment:
is TrimPrefix required here? We do `obj, err := fs.headObject(name)` some lines above `obj, err := fs.headObject(name+"/")` should work

Reply:
I'll remove the TrimPrefix if it is not required, thank you

Reply:
Testing right now (through the Minio Python client, connecting to B2 via its S3 API):
```py
import minio
c = minio.Minio(...)
# create a 0-bytes-keyed empty obj "testdir/"
c.put_object(
    bucket_name='test',
    object_name='testdir/',
    data=io.BytesIO(b''),
    length=0,
    content_type='inode/directory') # 'd41d8cd98f00b204e9800998ecf8427e'
# HEAD /<name>
c.stat_object('test', '/testdir') # NoSuchKey: message: The specified key does not exist.
# HEAD <name>
c.stat_object('test', 'testdir') # NoSuchKey: message: The specified key does not exist.
# HEAD <name>/
c.stat_object('test', 'testdir/') # <minio.definitions.Object at 0x7fea85112b90>
# cleanup
c.remove_object('test', 'testdir/')
# create a 0-bytes-keyed empty obj "/testdir/"
c.put_object(
    bucket_name='test',
    object_name='/testdir/',
    data=io.BytesIO(b''),
    length=0,
    content_type='inode/directory') # 'd41d8cd98f00b204e9800998ecf8427e'
# HEAD /<name>/
c.stat_object('test', '/testdir/') # <minio.definitions.Object at 0x7fea85112b90>
# HEAD /<name>
c.stat_object('test', '/testdir') # NoSuchKey: message: The specified key does not exist.
# HEAD <name>
c.stat_object('test', 'testdir') # NoSuchKey: message: The specified key does not exist.
# HEAD <name>/
c.stat_object('test', 'testdir/') # NoSuchKey: message: The specified key does not exist.
```
Both `/<name>/` and `name/` are valid, but at root `/<name>/` is considered a file rather than a directory, at least by the Backblaze web UI:

![2021-01-12-174804_702x160_scrot](https://user-images.githubusercontent.com/4138706/104345266-6d78ed00-54fe-11eb-959a-f397fba6a573.png)

Also, testing with directories created by sftpgo itself it seems they are created as `<name>/` rather than `/<name>/`, while `name` in this case starts with `/` (that's why I used `strings.TrimPrefix`).